tests/test_resolver.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: BSD-3-Clause
+import pytest
+from unittest import mock
+
+from operator_manifest.resolver import resolve_image_reference
+
+
+class TestResolver:
+    @mock.patch('subprocess.run')
+    def test_resolve_schema_version_2(self, subprocess_run):
+        subprocess_run.return_value = mock.Mock(returncode=0, stdout='{"schemaVersion": 2}')
+        # This is the digest of the manifest itself
+        expected = (
+            'example.com/foo/bar@'
+            'sha256:c5d902c53b4afcf32ad746fd9d696431650d3fbe8f7b10ca10519543fefd772c'
+        )
+        resolved = resolve_image_reference('example.com/foo/bar:latest')
+        assert resolved == expected
+        subprocess_run.assert_called_once()
+        assert '--raw' in subprocess_run.call_args[0][0]
+
+    @mock.patch('subprocess.run')
+    def test_resolve_schema_version_1(self, subprocess_run):
+        subprocess_run.side_effect = [
+            mock.Mock(returncode=0, stdout='{"schemaVersion": 1}'),
+            mock.Mock(returncode=0, stdout='{"Digest": "sha256:1"}'),
+        ]
+        # The digest determined by skopeo
+        expected = 'example.com/foo/bar@sha256:1'
+        resolved = resolve_image_reference('example.com/foo/bar:latest')
+        assert resolved == expected
+        assert subprocess_run.call_count == 2
+        # The first call to skopeo uses --raw
+        assert '--raw' in subprocess_run.call_args_list[0][0][0]
+        # The second call to skopeo does not use --raw and relies on skopeo to determine digest
+        assert '--raw' not in subprocess_run.call_args_list[1][0][0]
+
+    @mock.patch('subprocess.run')
+    def test_resolve_from_digest(self, subprocess_run):
+        subprocess_run.return_value = mock.Mock(returncode=0, stdout='{"schemaVersion": 2}')
+        # This is the digest of the manifest itself
+        image_reference = (
+            'example.com/foo/bar@'
+            'sha256:c5d902c53b4afcf32ad746fd9d696431650d3fbe8f7b10ca10519543fefd772c'
+        )
+        resolved = resolve_image_reference(image_reference)
+        assert resolved == image_reference
+        subprocess_run.assert_called_once()
+        assert '--raw' in subprocess_run.call_args[0][0]
```

Comment:
How is this different from `test_resolve_schema_version_2`?

Reply:
I added a comment to clarify. The difference is that `resolve_image_reference` is called with an already pinned image reference. Hmmm would it make it easier if it was a single test with `parametrize`?

Reply:
Ah, *that's* what I missed. Seems fine to me